core/src/main/java/hudson/PluginStrategy.java

```diff
@@ -92,5 +92,5 @@ PluginWrapper createPluginWrapper(File archive)
      * @param dependee newly loaded plugin.
      * @since 1.557
      */
-    void updateDependency(PluginWrapper depender, PluginWrapper dependee);
+    default void updateDependency(PluginWrapper depender, PluginWrapper dependee) {}
```

Comment:
Could add logging I suppose. Probably unimportant since this is SPI is not implemented by plugins, only by core.

I have thought about an `@Abstract` annotation with corresponding annotation processor, but never got to it.